internal/rbd/diskusage.go

```diff
@@ -24,6 +24,15 @@ import (
 // rbd_sparify() to free zero-filled blocks and reduce the storage consumption
 // of the image.
 func (ri *rbdImage) Sparsify() error {
+	inUse, err := ri.isInUse()
+	if err != nil {
+		return fmt.Errorf("failed to check if image is in use: %w", err)
+	}
+	if inUse {
+		// if the image is in use, we should not sparsify it, return ErrImageInUse.
+		return ErrImageInUse
```

Comment:
ot does not feel _balanced_ to return this error here, better have it returned by `ri.isInUse()` maybe?

Reply:
> ot does not feel _balanced_ to return this error here, better have it returned by `ri.isInUse()` maybe?

The `rbdImage.isInUse()`  is an internal function and I think its fine to return an error liked `ErrImageInUse` based on requirements of an exported function such as `Sparsify()` using the former internal function's output.

Reply:
ok!